### PR TITLE
feat: Respect remote servers configuration limits

### LINKED ISSUE #19: Limit batch sizes (respect server's /info/configuration limits)
The loadtester exceeds the [Spanner nodes' write limits](https://github.com/mozilla-services/syncstorage-rs/issues/298). It should limit its total batch size to a smaller amount than it currently produces.

We could hardcode it to a smaller limit for now, but the ideal solution would be to query the server's `/info/configuration` endpoint for its advertised max_total_records limit, and respect that value.

### PR DESCRIPTION
**Pull request by jrconlin**
also `assert` is the anti-christ, so make the errors more meaningful
than `(202,)`

Closes #19

### CODE REVIEW

File: loadtest.py [hunk 1]
```diff
@@ -69,19 +69,27 @@ def _run():
...
 
+    # Respect the server limits.
+    _, config = await storage.get("/info/configuration")
+    print("Config: {}".format(json.dumps(config, indent=3)))
+    # fix up consts
+    payload = _PAYLOAD[:config.get("max_record_payload_bytes")]
+    # GET requests to meta/global
+    num_requests = min(get_num_requests('metaglobal'),
+                       config.get("max_post_records"))
+    batch_max_count = min(BATCH_MAX_COUNT, config.get("max_total_records"))
```
Comment (pjenvey) [inline loadtest.py]:
```suggestion
    batch_max_count = min(_BATCH_MAX_COUNT, config.get("max_total_records"))
```

File: loadtest.py [hunk 2]
```diff
@@ -69,19 +69,27 @@ def _run():
 async def test(session):
     storage = session.storage
 
+    # Respect the server limits.
+    _, config = await storage.get("/info/configuration")
+    print("Config: {}".format(json.dumps(config, indent=3)))
```
Comment (pjenvey) [inline loadtest.py]: this is per each individual test run so this'll be too chatty, let's comment it out

File: storage/__init__.py [hunk 3]
```diff
@@ -222,12 +222,20 @@ def _auth(self, meth, url):
...
+                            "Reauth Response {} not in {}".format(
+                                resp.status,
+                                statuses)
+                        )
                     body = await resp.json()
                     return resp, body
             else:
                 if statuses is not None:
-                    assert resp.status in statuses, statuses
+                    assert resp.status in statuses,  (
```
Comment (pjenvey) [inline storage/__init__.py]: thanks! :cry: #tearsofjoy

File: loadtest.py (hunk 1 above)
Comment (jrconlin) [reply loadtest.py]: i... how the hell did that pass locally?
thanks

Comment (jrconlin) [reply loadtest.py]: ah, because, apparently, i was running the script on a different machine. awesome.

**Comment by tublitzed** [comment]
Hey, @jrconlin  - I removed this one from the board - let's just keep the issues (not the PRs) on the board, that way we don't have dupes of the same thing. (I moved the related issue #19 into review instead).

**Pull request merged**